app-android/src/main/java/com/ustadmobile/port/android/view/ReportFilterEditFragment.kt

```diff
@@ -272,4 +294,173 @@ class ReportFilterEditFragment : UstadEditFragment<ReportFilter>(), ReportFilter
+                enabled = uiState.fieldsEnabled,
+                keyboardOptions = KeyboardOptions(keyboardType = KeyboardType.Number),
+                onValueChange = {
+                    onReportFilterChanged(uiState.reportFilter?.shallowCopy{
+                        reportFilterValue = it
+                    })
+                }
+            )
+        }
+
+        item {
+            Row {
+                UstadTextEditField(
+                    modifier = Modifier.weight(0.5F),
+                    value = uiState.reportFilter?.reportFilterValueBetweenX ?: "",
+                    label = stringResource(id = R.string.from),
+                    error = uiState.valuesError,
+                    enabled = uiState.fieldsEnabled,
+                    keyboardOptions = KeyboardOptions(keyboardType = KeyboardType.Number),
+                    onValueChange = {
+                        onReportFilterChanged(uiState.reportFilter?.shallowCopy{
+                            reportFilterValueBetweenX = it
+                        })
+                    }
+                )
+
+                Spacer(modifier = Modifier.width(10.dp))
+
+                UstadTextEditField(
+                    modifier = Modifier.weight(0.5F),
+                    value = uiState.reportFilter?.reportFilterValueBetweenY ?: "",
+                    label = stringResource(id = R.string.toC),
+                    error = uiState.valuesError,
+                    enabled = uiState.fieldsEnabled,
+                    keyboardOptions = KeyboardOptions(keyboardType = KeyboardType.Number),
+                    onValueChange = {
+                        onReportFilterChanged(uiState.reportFilter?.shallowCopy{
+                            reportFilterValueBetweenY = it
+                        })
+                    }
+                )
+            }
+        }
+
+        items(
+            items = uiState.uidAndLabelList
+        ){ uidAndLabel ->
+            ListItem(
+                text = {
+                       TextButton(onClick = { onClickEditFilter(uidAndLabel)  }) {
+                           Text(uidAndLabel.labelName ?: "")
+                       }
+                },
+                trailing = {
+                    IconButton(onClick = { onClickRemoveFilter(uidAndLabel) }) {
+                        Icon(
+                            imageVector = Icons.Filled.Delete,
+                            contentDescription = stringResource(id = R.string.delete),
+                        )
+                    }
+                }
+            )
+        }
+
+        item {
+            ListItem(
+                modifier = Modifier.clickable {
+                    onClickNewItemFilter()
+                },
+                text = { Text(uiState.createNewFilter) },
+                icon = {
+                    Icon(
+                        imageVector = Icons.Filled.Add,
+                        contentDescription = stringResource(id = R.string.add),
```

Comment:
contentDescription here should be null or blank. There is already text for this item (e.g. it is not just an icon on it's own). The icon is just for appearance. 

We need the contentDescription on delete because there is no text for that clickable item (there is only the icon for the trash). For adding, we have text.